### PR TITLE
Use Gatesets in cirq/ion

### PR DESCRIPTION
**Pull request by tanujkhattar**
This PR adopts the use of `cirq.Gateset` in cirq/ion/*. Part of the roadmap item #3243

### CODE REVIEW

File: cirq-core/cirq/ion/ion_device.py [hunk 1]
```diff
@@ -28,6 +28,17 @@ class IonDevice(devices.Device):
     Qubits have all-to-all connectivity.
     """
 
+    gateset = ops.Gateset(
```
Comment (MichaelBroughton) [inline cirq-core/cirq/ion/ion_device.py:31]: Nit: can we avoid declaring this as a class level attribute (style guide discouraged) and move it into the `__init__` ?

**Review by MichaelBroughton — COMMENTED** [review]
LGTM just the one nit to be fixed then we can merge.

File: cirq-core/cirq/ion/ion_device.py (hunk 1 above)
Comment (tanujkhattar) [reply cirq-core/cirq/ion/ion_device.py:31]: sure, will do.

Comment (tanujkhattar) [reply cirq-core/cirq/ion/ion_device.py]: Added a module level helper to create the gateset and moved `gateset` to `__init__`.  LMK if this looks good.

**Review by MichaelBroughton — APPROVED** [review]

**Pull request merged**